app/controllers/index.js

```diff
@@ -58,58 +17,82 @@ export default Ember.Controller.extend({
       });
   },
 
-  _handleProposalCreated(proposalId, creatorAddress, recipientAddress, amount) {
-    if (Ember.isPresent(this.get('model.proposals')
-             .findBy('id', proposalId.toNumber()))) {
+  contributors: alias('model.contributors'),
+  contributorsWithKredits: filter('contributors', function(contributor) {
+    return contributor.get('balance') !== 0;
+  }),
+  contributorsSorting: ['balance:desc'],
+  contributorsSorted: sort('contributorsWithKredits', 'contributorsSorting'),
+
+  proposals: computed('model.proposals.[]', 'contributors.[]', function() {
+    return this.get('model.proposals')
+               .map((proposal) => {
+                 let contributor = this.get('contributors')
+                                       .findBy('id', proposal.get('recipientId'));
+                 if (contributor) {
+                   proposal.set('contributor', contributor);
+                 }
+
+                 return proposal;
+               });
+  }),
+  proposalsOpen: filterBy('proposals', 'isExecuted', false),
+  proposalsClosed: filterBy('proposals', 'isExecuted', true),
+  proposalsSorting: ['id:desc'],
+  proposalsClosedSorted: sort('proposalsClosed', 'proposalsSorting'),
+  proposalsOpenSorted: sort('proposalsOpen', 'proposalsSorting'),
+
+  _handleProposalCreated(proposalId, creatorAddress, recipientId, amount) {
+    // TODO: check if proposalId is already a string
+    let proposal = this.get('proposals')
+                       .findBy('id', proposalId.toString());
+    if (proposal) {
       Ember.Logger.debug('[index] proposal exists, not adding from event');
-      return false;
+      return;
     }
 
-    let proposal = Proposal.create({
-      id: proposalId.toNumber(),
+    // TODO: we should remove votesNeeded
+    proposal = this.get('kredits').buildModel('proposal', {
```

Comment:
we should reuse the `getProposalById` function here to load all the proposal data. 
adds another call but makes sure we have all the proposal data (and not only the data from the event)